### PR TITLE
Moved some tests to new annotations

### PR DESCRIPTION
**Pull request by ruwanta**
### Proposed changes in this pull request

- Some tests to annotation

### When should this PR be merged

ASAP

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandlerTest.java [hunk 1]
```diff
@@ -18,332 +18,210 @@
...
-        mockStatic(OAuthServerConfiguration.class);
-        when(OAuthServerConfiguration.getInstance()).thenReturn(serverConfiguration);
-        when(serverConfiguration.getCallbackHandlerMetaData()).thenReturn(callbackHandlerMetaData);
 
-        OAuthCallbackManager oAuthCallbackManager = new OAuthCallbackManager();
-        Field field = AbstractAuthorizationGrantHandler.class.getDeclaredField("callbackManager");
-        field.setAccessible(true);
-        field.set(handler, oAuthCallbackManager);
-        field.setAccessible(false);
+        //        OAuthCallbackManager oAuthCallbackManager = new OAuthCallbackManager();
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandlerTest.java]: Shall we remove the commented out code?

File: test-utils/org.wso2.carbon.identity.oauth.common.testng/src/main/java/org/wso2/carbon/identity/test/common/testng/utils/WhiteBox.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.test.common.testng.utils;
+
+import java.lang.reflect.Method;
+
+/**
+ * Some White-Box utilities, to call/set internal states with reflection.
+ * Mockito and PowerMockito has Witebox class. But seems not quite simple.
```
Comment (mefarazath) [inline test-utils/org.wso2.carbon.identity.oauth.common.testng/src/main/java/org/wso2/carbon/identity/test/common/testng/utils/WhiteBox.java]: Witebox --> Whitebox

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandlerTest.java [hunk 3]
```diff
@@ -18,156 +18,80 @@
...
-        OauthTokenIssuer oAuthIssuer = PowerMockito.mock(OauthTokenIssuer.class);
-
-        Field f3 = baseClass.getClass().getSuperclass().getDeclaredField("oauthIssuerImpl");
-        f3.setAccessible(true);
-        f3.set(baseClass, oAuthIssuer);
+        new OAuthAppDAO().addOAuthApplication(oAuthAppDO);
 
-        PowerMockito.when(oAuthIssuer.accessToken(authAuthzReqMessageContext)).thenReturn("1234567890");
-        PowerMockito.when(oAuthIssuer.refreshToken(authAuthzReqMessageContext)).thenReturn("1234567890");
+        //        PowerMockito.when(OAuth2Util
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandlerTest.java]: Shall we remove commented code?

Comment (ruwanta) [reply components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandlerTest.java]: +1 will do

**Pull request merged**